File: update/update.php
<?php
/**
 * Updates the property of the associative array using the function passed.
 * 
 * @param Array $arr - associative array to update.
 * @param String $key - key of property to be updated.
 * @param Function $func - function that updates the property.
 * 
 * @return Array Updated array.
 */
function update(array $arr, string $key, callable $func): array {
    if (!is_null($arr)) {
        $arr[$key] = $func($arr[$key]);
    }
    return $arr;
}

/**
 * Updates a property that's inside a nested associative array.
 * 
 * @param Array $arr - associative array to update.
 * @param Array $keys - Array representing path to nested property in $arr passed.
 * @param Function $func - function that updates the property.
 * 
 * @return Array Updated array.
 */
function nestedUpdate(array $arr, array $keys, callable $func): array {
    if (count($keys) > 1) {
        return nestedUpdate($arr[$keys[0]], array_slice($keys, 1), $func);
    } else if (count($keys) == 1) {
        return update($arr, $keys[0], $func);
    } else {
        throw new Exception("atleast one key must be passed");
    }
}

File: update/update.py
"""Alters the key value for the dictionary passed using the function passed.

    Arguments:
        d (dictionary): The dictionary to be updated.
        key (string): The key to the value to be updated.
        func (function): The function that alters the value of the key passed.
    
    Returns:
        dictionary: The new dictionary.
"""
def update(d, key, func):
    if d is None:
        return d
    else:
        d[key] = func(d[key])
        return d

"""Alters the value when it's inside a nested dictionary.
    Arguments:
        d (dictionary): The dictionary to be updated.
        key (list): A list of strings representing the path to the value to be altered.
        func (function): The function that alters the value of the key passed.
    
    Returns:
        dictionary: The new dictionary.
"""
def nestedUpdate(d, keys, func):
    if len(keys) < 1:
        raise Exception("Atleast one key is required")
    elif (len(keys) == 1):
        update(d, keys[0], func)
        return d
    else:
        return nestedUpdate(d[keys[0]], keys[1:], func)
    return